crates/sui-framework/deps/move-stdlib/sources/u64.move

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Basic math for nicer programmability
+module std::u64 {
+    use std::ascii;
+    use std::option::{Self, Option};
+    use std::string;
+    use std::vector;
+
+    /// Largest possible u64 value
+    const MAX: u64 = 0;
+
+    /// Attempting to perform an invalid operation (e.g., mean) on an empty vector
+    const EVectorEmpty: u64 = 0;
+
+    /// Return the larger of `x` and `y`
+    public fun max(x: u64, y: u64): u64 {
+        if (x > y) {
+            x
+        } else {
+            y
+        }
+    }
+
+    /// Return the smaller of `x` and `y`
+    public fun min(x: u64, y: u64): u64 {
+        if (x < y) {
+            x
+        } else {
+            y
+        }
+    }
+
+    /// Return the value of a base raised to a power.
+    /// This function is not overflow-safe and will abort with an overflow error if `base^exponent` > `MAX`
```

Comment:
This seems worth writing a `spec` for -- can we annotate this comment some way to make such things easy to find? cc @awelc